homeassistant/components/swiss_public_transport/__init__.py

```diff
@@ -65,3 +66,51 @@ async def async_unload_entry(
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+async def async_migrate_entry(
+    hass: core.HomeAssistant, config_entry: config_entries.ConfigEntry
+) -> bool:
+    """Migrate config entry."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    if config_entry.minor_version > 3:
+        # This means the user has downgraded from a future version
+        return False
+
+    if config_entry.minor_version == 1:
+        # Remove wrongly registered devices and entries
+        new_unique_id = (
+            f"{config_entry.data[CONF_START]} {config_entry.data[CONF_DESTINATION]}"
+        )
+        entity_registry = er.async_get(hass)
+        device_registry = dr.async_get(hass)
+        device_entries = dr.async_entries_for_config_entry(
+            device_registry, config_entry_id=config_entry.entry_id
+        )
+        for dev in device_entries:
+            device_registry.async_remove_device(dev.id)
+
+        entity_id = entity_registry.async_get_entity_id(
+            Platform.SENSOR, DOMAIN, "None_departure"
+        )
+        if entity_id:
+            entity_registry.async_update_entity(
+                entity_id=entity_id,
+                new_unique_id=f"{new_unique_id}_departure",
+            )
+            _LOGGER.debug(
+                "Faulty entity with unique_id 'None_departure' migrated to new unique_id '%s'",
+                f"{new_unique_id}_departure",
+            )
+
+        # Set a valid unique id for config entries
+        config_entry.unique_id = new_unique_id
+        config_entry.minor_version = 2
+        hass.config_entries.async_update_entry(config_entry)
+
+    _LOGGER.debug(
+        "Migration to minor version %s successful", config_entry.minor_version
```

Comment:
If the migration changes in the future, this log message may not make sense anymore. I'd log both major and minor version.